tests/unit/amo/components/TestAddonHead.js

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+
+import AddonHead, { AddonHeadBase } from 'amo/components/AddonHead';
+import {
+  ADDON_TYPE_COMPLETE_THEME,
+  ADDON_TYPE_DICT,
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_LANG,
+  ADDON_TYPE_OPENSEARCH,
+  ADDON_TYPE_STATIC_THEME,
+  ADDON_TYPE_THEME,
+  CLIENT_APP_ANDROID,
+  CLIENT_APP_FIREFOX,
+} from 'core/constants';
+import { createInternalAddon } from 'core/reducers/addons';
+import {
+  dispatchClientMetadata,
+  fakeAddon,
+  fakeTheme,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  const render = ({
+    store = dispatchClientMetadata().store,
+    ...props
+  } = {}) => {
+    return shallowUntilTarget(
+      <AddonHead i18n={fakeI18n()} store={store} {...props} />,
+      AddonHeadBase,
+    );
+  };
+
+  it('renders nothing when no add-on is passed', () => {
+    const root = render();
+
+    expect(root.find(Helmet)).toHaveLength(0);
+  });
+
+  it.each([
+    [ADDON_TYPE_DICT, 'Dictionary'],
+    [ADDON_TYPE_EXTENSION, 'Extension'],
+    [ADDON_TYPE_LANG, 'Language Pack'],
+    [ADDON_TYPE_OPENSEARCH, 'Search Tool'],
+    [ADDON_TYPE_STATIC_THEME, 'Theme'],
+    [ADDON_TYPE_THEME, 'Theme'],
+    [ADDON_TYPE_COMPLETE_THEME, 'Add-on'],
+  ])('renders an HTML title for Firefox (add-on type: %s)', (type, name) => {
+    const lang = 'fr';
+    const addon = createInternalAddon({ ...fakeAddon, type });
+    const { store } = dispatchClientMetadata({
+      clientApp: CLIENT_APP_FIREFOX,
+      lang,
+    });
+    const root = render({ addon, store });
+
+    expect(root.find('title')).toHaveText(
+      `${addon.name} – Get this ${name} for 🦊 Firefox (${lang})`,
+    );
+  });
+
+  it.each([
```

Comment:
Nit: It might make for a more messy test, but you could have 2 nested `it.each` calls, one which loops through the 2 values of `clientApp` and another that loops through the values of `type`, instead of two complete tests for `CLIENT_APP_FIREFOX` and `CLIENT_APP_ANDROID`. It would remove a lot of duplication, but it might be a bit ugly.

Reply:
ah, that makes me think about a matrix instead:

```
[
  [CLIENT_APP_FIREFOX, ADDON_TYPE_DICT, 'Dictionary'],
  ...,
  [CLIENT_APP_ANDROID, ADDON_TYPE_DICT, 'Dictionary'],
  ...,
]
```

Reply:
The thing is we would need the title somewhere, mmh that's tricky actually.